### PR TITLE
Add PSR16 caching

### PR DESCRIPTION
**Pull request by javabudd**
This PR adds PSR16 caching and bypasses enforcing argument types by returning an InvalidArgumentException as defined in the PSR CacheInterface

### CODE REVIEW

File: src/Cache/Service/Warm.php [hunk 1]
```diff
@@ -6,29 +6,28 @@
 
 namespace Jstewmc\Gravity\Cache\Service;
 
-use Jstewmc\Gravity\Cache\Data\Cache;
 use Jstewmc\Gravity\{Deprecation, Id, Path, Service};
 use Psr\Log\LoggerInterface as Logger;
+use Psr\SimpleCache\CacheInterface;
```
Comment (jstewmc) [inline src/Cache/Service/Warm.php]: We could use - `use Psr\SimpleCache\CacheInterface as Cache` - for a little more brevity. I did the same with the `Psr\Log\LoggerInterface as Logger` (at least when I realized I could haha).

File: src/Manager/Data/Manager.php [hunk 2]
```diff
@@ -20,14 +19,19 @@
 
 class Manager
 {
+    /** @var GetId */
```
Comment (jstewmc) [inline src/Manager/Data/Manager.php:22]: I got so into removing all my superfluous comments that I forgot to include these types for object properties. Kinda jumped right to the other side of the spectrum haha.

**Comment by jstewmc** [comment]
Closes #5.

**Comment by javabudd** [comment]
This is still a WIP, I opened this PR up a bit early because I wanted to use travis for tests.

File: src/Cache/Service/Warm.php (hunk 1 above)
Comment (javabudd) [reply src/Cache/Service/Warm.php]: I am a big fan of aliasing imports. I wasn't sure what you preferred but +1

File: src/Manager/Data/Manager.php (hunk 2 above)
Comment (javabudd) [reply src/Manager/Data/Manager.php:22]: I honestly thought you were doing this so a conversion to typed properties would be easier in PHP 7.4 (https://wiki.php.net/rfc/typed_properties_v2)

I like the idea of not having them, it's just a little easier in the meantime for IDE's

File: src/Path/Data/Service.php [hunk 3]
```diff
@@ -8,5 +8,8 @@
 
 class Service extends Path
 {
-    public const SEPARATOR = '\\';
```
Comment (jstewmc) [inline src/Path/Data/Service.php:11]: Why the switch from constants to static functions?

**Pull request merged**